packages/js/core/src/types/Client.ts

```diff
@@ -15,13 +15,17 @@ import { UriResolverHandler } from "./UriResolver";
 import { WrapManifest } from "@polywrap/wrap-manifest-types-js";
 import { Result } from "@polywrap/result";
 
-export interface ClientConfig<TUri extends Uri | string = string> {
-  redirects: UriRedirect<TUri>[];
-  plugins: PluginRegistration<TUri>[];
-  interfaces: InterfaceImplementations<TUri>[];
-  envs: Env<TUri>[];
-  resolver: IUriResolver<unknown>;
-}
+type ClientConfigInternal<TUri extends Uri | string = string> = {
```

Comment:
Also, we use `interface` elsewhere in the client and core. I think it makes sense for us to maintain the same code style throughout, even if that means we can't use the `Readonly<>` syntax.

```typescript
export interface ClientConfig<TUri extends Uri | string = string> {
  readonly redirects: readonly UriRedirect<TUri>[];
  readonly plugins: readonly PluginRegistration<TUri>[];
  readonly interfaces: readonly InterfaceImplementations<TUri>[];
  readonly envs: readonly Env<TUri>[];
  readonly resolver: Readonly<IUriResolver<unknown>>;
}
```

Reply:
Regarding your stylistic preference, your approach is a strictly better solution, and I'm going with it. The style I have here right now is mostly a remnant of some experimentation from before. TBH I'm amazed that it didn't come to mind when I was working on this...

There are notably two issues with `interface` which led me to change it over to `type`:
- First, as you mentioned, is that we can't make an interface `readonly` by definition. This would allow for something like the following to be valid code:
    ```typescript
    const a = new ClientConfigBuilder().addDefaults().build();

    a.envs = [];
    ```
    Although `envs` is a `ReadOnlyArray<Env>`, the `envs` property itself can be reassigned.    I wanted to prevent this, and using `type` allowed me to neatly solve this.
- I have attempted to export an `interface` which `extends` an internal interface (which is not exported). 
    This syntax worked, but produced a linter error (a valid one - `no-empty-interface`):
    ```typescript
    export interface Interface extends Readonly<InterfaceInternal> {};
    ```
    However, when fixing the lint error with:
    ```typescript
    export interface Interface extends Readonly<InterfaceInternal>;
    ```
    The compiler was having trouble with determining the underlying type of the interface (since `InterfaceInternal` was not exported).
    I've gone over some issues on the TS GitHub, and while there are some TS settings that can be fiddled with (amongst a lithany of hacks that also "make it work"), I've kind of figured that this is either a bug or simply not supported. 
    I've come to the conclusion that  I'd rather avoid using interfaces overall in this case, especially as I believe that, in this case, `interface` and `type` are essentially interchangeable, with the only real difference being that interfaces support declaration merging (something like `partial` in C#), which is an anti-pattern in my eyes.

Sorry for the long reply, I really wanted to give context to this and I'd love to discuss this further if you have any ideas or simply disagree with me! :smile:

Reply:
The `Readonly<>` type wrapper is the same as adding `readonly` to each top-level property. The `Readonly<>` type is just syntax sugar. So the interface example I shared is equivalent.

https://www.typescriptlang.org/docs/handbook/utility-types.html#readonlytype